Userland/Applications/FileManager/DirectoryView.cpp

```diff
@@ -37,8 +37,8 @@ void spawn_terminal(GUI::Window* window, StringView directory)
 NonnullRefPtr<GUI::Action> LauncherHandler::create_launch_action(Function<void(LauncherHandler const&)> launch_handler)
 {
     auto icon = GUI::FileIconProvider::icon_for_executable(details().executable).bitmap_for_size(16);
-    return GUI::Action::create(details().name, move(icon), [this, launch_handler = move(launch_handler)](auto&) {
-        launch_handler(*this);
+    return GUI::Action::create(details().name, move(icon), [strong_this = NonnullRefPtr(*this), launch_handler = move(launch_handler)](auto&) {
```

Comment:
I'm not sure to understand how this change is related to the `Vector`s. If it's not, it would be nice to split the change in two commits. Otherwise, a small explanation in the commit message will do it.

Reply:
Updated the commit message, but now I wonder if an OwnPtr could be a better fit for a LauncherHandler 🤔

Reply:
Thanks for the explanation!